crowbar_framework/app/models/api/node.rb

```diff
@@ -15,6 +15,61 @@
 #
 
 module Api
-  class Node
+  class Node < Tableless
+    def repocheck(options = {})
+      addon = options.fetch(:addon, "os")
+      features = []
+      features.push(addon)
+
+      {}.tap do |ret|
+        ret[addon] = {
+          "available" => true
+        }
+        platform = Api::Upgrade.new.target_platform(platform_exception: addon)
+
+        features.each do |feature|
+          if node_architectures(addon: addon).any?
+            node_architectures(addon: addon).each do |architecture|
+              unless ::Crowbar::Repository.provided_and_enabled?(feature,
+                                                                 platform,
+                                                                 architecture)
+                ::Openstack::Upgrade.enable_repos_for_feature(feature, Rails.logger)
+              end
+              available, repolist = ::Crowbar::Repository.provided_and_enabled_with_repolist(
+                feature, platform, architecture
+              )
+              ret[addon]["available"] &&= available
+              ret[addon]["repos"] ||= {}
+              ret[addon]["repos"].deep_merge!(repolist.deep_stringify_keys)
+            end
+          else
+            ret.delete[addon]
+          end
+        end
+      end
+    end
+
+    protected
+
+    def node_architectures(options = {})
```

Comment:
@vuntz how can we determine the architectures for the `os` and the `openstack` feature? i just realized, we can't do the same check (by one deployed barclamp) as here in `node_architectures`